partner_firstname/models/res_partner.py

```diff
@@ -0,0 +1,271 @@
+            record.name = clean
+            record._inverse_name()
+
+    @api.model
+    def _get_whitespace_cleaned_name(self, name, comma=False):
+        """Remove redundant whitespace from :param:`name`.
+
+        Removes leading, trailing and duplicated whitespace.
+        """
+        if isinstance(name, bytes):
+            # With users coming from LDAP, name can be a byte encoded string.
+            # This happens with FreeIPA for instance.
+            name = name.decode("utf-8")
+
+        try:
+            name = " ".join(name.split()) if name else name
+        except UnicodeDecodeError:
+            # with users coming from LDAP, name can be a str encoded as utf-8
+            # this happens with ActiveDirectory for instance, and in that case
+            # we get a UnicodeDecodeError during the automatic ASCII -> Unicode
+            # conversion that Python does for us.
+            # In that case we need to manually decode the string to get a
+            # proper unicode string.
+            name = " ".join(name.decode("utf-8").split()) if name else name
+
+        if comma:
+            name = name.replace(" ,", ",")
+            name = name.replace(", ", ",")
+        return name
+
+    @api.model
+    def _get_inverse_name(self, name, is_company=False):
+        """Compute the inverted name.
+
+        - If the partner is a company, save it in the lastname.
+        - Otherwise, make a guess.
+
+        This method can be easily overriden by other submodules.
+        You can also override this method to change the order of name's
+        attributes
+
+        When this method is called, :attr:`~.name` already has unified and
+        trimmed whitespace.
+        """
+        # Company name goes to the lastname
+        if is_company or not name:
+            parts = [name or False, False]
+        # Guess name splitting
+        else:
+            order = self._get_names_order()
+            # Remove redundant spaces
+            name = self._get_whitespace_cleaned_name(
+                name, comma=(order == "last_first_comma")
+            )
+            parts = name.split("," if order == "last_first_comma" else " ", 1)
+            if len(parts) > 1:
+                if order == "first_last":
+                    parts = [" ".join(parts[1:]), parts[0]]
+                else:
+                    parts = [parts[0], " ".join(parts[1:])]
+            else:
+                while len(parts) < 2:
+                    parts.append(False)
+        return {"lastname": parts[0], "firstname": parts[1]}
+
+    def _inverse_name(self):
+        """Try to revert the effect of :meth:`._compute_name`."""
+        for record in self:
+            parts = record._get_inverse_name(record.name, record.is_company)
+            record.lastname = parts["lastname"]
+            record.firstname = parts["firstname"]
+
+    @api.constrains("firstname", "lastname")
+    def _check_name(self):
```

Comment:
`_check_name` is already in use. https://github.com/odoo/odoo/blob/ea01165d9486572269c44597a1db49b31bf8aba3/odoo/addons/base/models/res_partner.py#L326
This will also fix the unittest issues.

```suggestion
    def _check_firstname_lastname(self):
```

Reply:
@vincent-hatakeyama i did first review on the wrong PR (#2179). 

The `_check_name` wasn't overwritten in previous version there was always the sql constraint `check_name`. 
But never a `api.constrains`. In v19 sql constraints are defined in a different way. Like you can see it. In the provided link above. The odoo default attribute `_check_name` clashes now with the method. Therefor a simple rename is enough.

Reply:
thanks @mt-software-de  for your explanation :) i will change that this morning

@legalsylvain  sorry for the late update, sure. i shall cherry pick missing commit right ? (sorry i'm not comfortable with this kind of operation) . 

i will do it today